lib/main.js

```diff
@@ -346,47 +346,48 @@ const Hydrogen = {
   },
 
   run(moveDown: boolean = false) {
-    if (!store.editor) return;
-    const codeBlock = codeManager.findCodeBlock();
+    const editor = store.editor;
+    if (!editor) return;
+    const codeBlock = codeManager.findCodeBlock(editor);
     if (!codeBlock) {
       return;
     }
 
     const [code, row] = codeBlock;
     if (code) {
       if (moveDown === true) {
-        codeManager.moveDown(row);
+        codeManager.moveDown(editor, row);
       }
       this.createResultBubble(code, row);
     }
   },
 
   runAll() {
-    if (!store.editor) return;
-    if (isMultilanguageGrammar(store.editor.getGrammar())) {
-      atom.notifications.addError(
-        '"Run All" is not supported for this file type!'
-      );
-      return;
-    }
-    if (store.kernel) {
-      this._runAll(store.kernel);
+    const editor = store.editor;
+    if (editor && store.kernel) {
+      if (isMultilanguageGrammar(editor.getGrammar())) {
```

Comment:
I would keep the check for `isMultilanguageGrammar` before the check for `kernel` to ensure a error message.